vmc/data_source_vmc_sddc.go

```diff
@@ -0,0 +1,201 @@
+	"github.com/hashicorp/terraform/helper/validation"
+	"github.com/vmware/vsphere-automation-sdk-go/lib/vapi/std/errors"
+)
+
+func dataSourceVmcSddc() *schema.Resource {
+	return &schema.Resource{
+		Read: dataSourceVmcSddcRead,
+
+		Schema: map[string]*schema.Schema{
+			"storage_capacity": {
+				Type:     schema.TypeString,
+				Optional: true,
+				ForceNew: true,
+				ValidateFunc: validation.StringInSlice([]string{
+					"15TB", "20TB", "25TB", "30TB", "35TB"}, false),
+			},
+			"sddc_name": {
+				Type:         schema.TypeString,
+				Required:     true,
+				ValidateFunc: validation.NoZeroValues,
+			},
+			"account_link_sddc_config": {
+				Type: schema.TypeList,
+				Elem: &schema.Resource{
+					Schema: map[string]*schema.Schema{
+						"customer_subnet_ids": {
+							Type: schema.TypeList,
+							Elem: &schema.Schema{
+								Type: schema.TypeString,
+							},
+							Optional: true,
+						},
+						"connected_account_id": {
+							Type:     schema.TypeString,
+							Optional: true,
+						},
+					},
+				},
+				Optional: true,
+				ForceNew: true,
+			},
+			"vpc_cidr": {
+				Type:     schema.TypeString,
+				Optional: true,
+				ForceNew: true,
+			},
+			"num_host": {
+				Type:         schema.TypeInt,
+				Required:     true,
+				ValidateFunc: validation.IntAtLeast(1),
+			},
+			"sddc_type": {
+				Type:     schema.TypeString,
+				Optional: true,
+				ForceNew: true,
+			},
+			"vxlan_subnet": {
+				Type:     schema.TypeString,
+				Optional: true,
+				ForceNew: true,
+			},
+			// TODO check the deprecation statement
+			"delay_account_link": {
+				Type:     schema.TypeBool,
+				Optional: true,
+				Default:  false,
+				ForceNew: true,
+			},
+			"provider_type": {
+				Type:     schema.TypeString,
+				Optional: true,
+				ForceNew: true,
+				Default:  "AWS",
+				ValidateFunc: validation.StringInSlice([]string{
+					"AWS", "ZEROCLOUD"}, false),
+			},
+			"skip_creating_vxlan": {
+				Type:     schema.TypeBool,
+				Optional: true,
+				Default:  true,
+				ForceNew: true,
+			},
+			"sso_domain": {
+				Type:     schema.TypeString,
+				Optional: true,
+				ForceNew: true,
+				Default:  "vmc.local",
+			},
+			"sddc_template_id": {
+				Type:     schema.TypeString,
+				Optional: true,
+				ForceNew: true,
+			},
+			"deployment_type": {
+				Type:     schema.TypeString,
+				Optional: true,
+				ForceNew: true,
+				Default:  "SingleAZ",
+				ValidateFunc: validation.StringInSlice([]string{
+					"SingleAZ", "MultiAZ",
+				}, false),
+			},
+			"region": {
+				Type:     schema.TypeString,
+				Required: true,
+				ForceNew: true,
+				ValidateFunc: validation.All(
+					validation.NoZeroValues,
+				),
+			},
+			"cluster_id": {
+				Type:     schema.TypeString,
+				Computed: true,
+			},
+			"host_instance_type": {
+				Type:     schema.TypeString,
+				Optional: true,
+				ForceNew: true,
+				ValidateFunc: validation.StringInSlice(
+					[]string{HostInstancetypeI3, HostInstancetypeR5}, false),
```

Comment:
Are still supporting R5?

Reply:
Its the initial implementation that hasn't been modified. Why do you ask?

Reply:
I don’t think we support R5 anymore. Its irrelevant to this MR. So, no need to act on it in this MR. But, please check with VMC team.